### PR TITLE
Add a temporary new class DefaultMixedNewAPI for cleaner development

### PR DESCRIPTION
**Pull request by JerryChen97**
**Context:**
In our Epic of porting `DefaultMixed` to new API standard, we realized #https://github.com/PennyLaneAI/pennylane/pull/6601/#pullrequestreview-2446682427 that direct changes to the original src are potentially risky since the legacy version and the new version are drastically different. Here we introduce a temporary name `DefaultMixedNewAPI` implemented as a second class inside the same `qml.devices.default_mixed` serving as a temporary class for holding the implementation of `preprocess` and `execute` of the new API of mixed qubit.

**Description of the Change:**
A new class within `qml.devices.default_mixed` named `DefaultMixedNewAPI`:
 - similar as qutrit mixed counterpart, see `qml.devices.default_qutrit_mixed`
 - preserves some input arguments from DefaultMixed, except for `analytic`, which accroding to the docs has already been deprecated
 - a dumb `execute` to avoid abstractclass error; will be implemented in other branch
 - a `support_derivatives` to check the supported diff method.
 - some imports are separated lined and the isort checks within this file disabled.

**Benefits:**
Better separation from legacy code.

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-78775]

### CODE REVIEW

**Comment by JerryChen97** [comment]
Considering that this is a temporary class and might need to be deleted at some point I prefer not including it inside the changelog.

File: pennylane/devices/default_mixed.py [hunk 1]
```diff
@@ -796,3 +805,107 @@ def apply(self, operations, rotations=None, **kwargs):
...
+        if execution_config is None or execution_config.gradient_method in {"backprop", "best"}:
+            return circuit is None or not circuit.shots
+        return False
+
+    @debug_logger
+    def execute(
+        self,
+        circuits: QuantumScript,
+        execution_config: Optional[ExecutionConfig] = None,
+    ) -> None:
```
Comment (JerryChen97) [inline pennylane/devices/default_mixed.py:910]:
```suggestion
    ) -> None:
        # pragma: no cover
```

File: pennylane/devices/default_mixed.py [hunk 2]
```diff
@@ -796,3 +805,108 @@ def apply(self, operations, rotations=None, **kwargs):
...
+        c_dtype (numpy.dtype): Complex datatype to use for computations. Default is np.complex128.
+        readout_prob (float): Probability of readout error for qubit measurements. Must be in :math:`[0,1]`.
+    """
+
+    _device_options = ("rng", "prng_key")  # tuple of string names for all the device options.
+
+    @property
+    def name(self):
+        """The name of the device."""
+        return "default.qubit.mixed"
```
Comment (albi3ro) [inline pennylane/devices/default_mixed.py:839]:
```suggestion
        return "default.mixed"
```

File: pennylane/devices/default_mixed.py [hunk 3]
```diff
@@ -796,3 +805,108 @@ def apply(self, operations, rotations=None, **kwargs):
...
+    # pylint: disable=too-many-positional-arguments
+    @debug_logger_init
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        wires=None,
+        shots=None,
+        seed="global",
+        # The following parameters are inherited from DefaultMixed
+        r_dtype=np.float64,
+        c_dtype=np.complex128,
```
Comment (albi3ro) [inline pennylane/devices/default_mixed.py:850]: Not sure if it's worth having these at the moment.  We do have a very backlogged epic to improve float32 support, but I'm not sure that this is worth it for `default.mixed` right now.

**Review by astralcai — APPROVED** [review]
Looks good.

File: codecov.yml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 ignore:
+  - "pennylane/devices/default_mixed.py"
```
Comment (albi3ro) [inline codecov.yml:2]: I'd prefer not to update this file, as I didn't even know it existed, so we probably won't remember to come back here. 

If we are having issues with the `execute` method, we can just do:
```
def execute(self, circuits, execution_config=None):
    raise NotImplementedError
```

And then
```
with pytest.raises(NotImplementedError):
     dev.execute(QuantumScript())
```

**Review by albi3ro — APPROVED** [review]
Thanks for your patience :) Looks good 👍

**Pull request merged**